hack/gen_rvar_table.py

```diff
@@ -0,0 +1,119 @@
+#!/bin/python
+
+import re
+
+with open('defaults/main.yml') as f:
+    lines = f.read().split("\n")
+
+def print_rvar_table(rvar_table):
+    keys = ["name", "description", "default", "required"]
+
+    sizes = {}
+    for key in keys:
+        sizes[key] = len(key)
+
+    for item in rvar_table:
+        for key in item.keys():
+            sizes[key] = max(sizes[key], len(item[key]))
+
+    headers = []
+    delimiters = []
+    for key in keys:
+        headers.append(" {}{} ".format(key, " "*(sizes[key]-len(key))))
+        delimiters.append("-{}-".format("-"*(sizes[key])))
+    print "|{}|".format("|".join(headers))
+    print "|{}|".format("|".join(delimiters))
+
+    # sort items by variable name
+    ki_items = {}
+    for item in rvar_table:
+        ki_items[item["name"]] = item
+
+    for key in sorted(ki_items):
+        item = ki_items[key]
+        data = []
+        for key in keys:
+            data.append(" {}{} ".format(item[key], " "*(sizes[key]-len(item[key]))))
+        print "|{}|".format("|".join(data))
+
+def get_delims(line, skip_list=False):
+    # replace all "...", '...' and [...] with # so it is easy to determine
+    # position of the ',' symbol
+    line = re.sub(r'("[^"]*")', lambda l: '#'*len(l.group()) , line)
+    line = re.sub(r"('[^']*')", lambda l: '#'*len(l.group()) , line)
+    if not skip_list:
+        line = re.sub(r"(\[[^\]]*\])", lambda l: '#'*len(l.group()) , line)
+    return re.findall(r"[^,]+", line)
+
+def parse_list(value):
```

Comment:
Additionally, it makes ordering less important. The doc strings just need to be in the parsed file somewhere. Or even in their own file. (I can see why that might not be ideal, however)